port/linux/omrosdump_helpers.c

```diff
@@ -206,30 +214,45 @@ renameDump(struct OMRPortLibrary *portLibrary, char *filename, pid_t pid, int si
 
 	if (0 != createCoreFileNameRC) {
 		/* we could not derive the name from core_pattern */
-		portLibrary->str_printf(portLibrary,
-								filename,
-								EsMaxPath,
-								"Unable to derive system core filename from pattern in %s: \"%s\". "
-								"%s"
-								" Look for core file generated by child process with pid = %i",
-								corePatternFileName,
-								corePatternFormat,
-								derivedAbsoluteCorePath,
-								pid);
+		portLibrary->str_printf(
+				portLibrary,
+				filename,
+				EsMaxPath,
+				"Unable to derive system core filename from pattern in %s: \"%s\". "
+				"%s"
+				" Look for core file generated by child process with pid = %i",
+				corePatternFileName,
+				corePatternFormat,
+				derivedAbsoluteCorePath,
+				pid);
 		return 1;
 	}
 
 	/* wait until the core file is created, or timeout */
 	waitCoreRC = waitCore(derivedAbsoluteCorePath);
 	if (0 != waitCoreRC) {
 		/* the core file does not exist, bail */
-		portLibrary->str_printf(portLibrary,
-								filename,
-								EsMaxPath,
-								"The core file created by child process with pid = %i was not found. Expected to find core file with name \"%s\"",
-								pid,
-								derivedAbsoluteCorePath);
-		return 1;
+		if (corePatternPiped) {
+			/* Nothing inherently wrong with using a core pipe program, so direct the user to its man page. */
+			filename[0] = '\0';
+			portLibrary->nls_printf(
+					portLibrary,
+					J9NLS_INFO | J9NLS_STDERR,
+					J9NLS_PORT_LINUXDUMP_PIPE_CORE_NOT_FOUND,
+					pid,
+					corePatternFileName,
+					corePatternFormatOriginal);
+			return 0;
```

Comment:
Returning zero in this situation seems wrong; the documented contract of this function is that `filename` will contain the name of the file created (either specified by the caller or computed here), but instead we will answer the empty string which will mislead users (see [gphandle.c](https://github.com/eclipse-openj9/openj9/blob/master/runtime/vm/gphandle.c#L906)) and break tests in [omrdumpTest.cpp](https://github.com/eclipse/omr/blob/master/fvtest/porttest/omrdumpTest.cpp#L167) and [j9dumpTest.c](https://github.com/eclipse-openj9/openj9/blob/master/runtime/tests/port/j9dumpTest.c#L205).

Reply:
Ok, that makes sense. This means that a piped `core_pattern` will lead to a `JVMDUMP012E` message instead of `JVMDUMP010I`, but that should be okay as `JVMPORT049I` will [document](https://github.com/eclipse-openj9/openj9-docs/issues/1033) the behavior in detail.